kernel/mem_domain.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <kernel.h>
+#include <kernel_structs.h>
+#if defined(CONFIG_ARM_CORE_MPU)
+#include <arch/arm/cortex_m/mpu/arm_core_mpu.h>
+#endif
+
+
+static u8_t max_partitions;
+
+
+static inline u8_t arch_get_max_partitions(void)
```

Comment:
I want a different approach here.
Declare the interfaces you need arch code to implement for memory domains in kernel/include/nano_internal.h
Then move all these arch_ functions under arch/arm/.
Their names should be prefixed with an underscore since they are private to the kernel.
We will have alternate implementations under arch/x86.

Reply:
revised.